sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -573,30 +573,79 @@ object GpuCast extends Arm {
       inclusiveMax: Boolean = true,
       errorMsg:String = GpuCast.OVERFLOW_MESSAGE): Unit = {
 
-    def throwIfAny(cv: ColumnView): Unit = {
-      withResource(cv) { cv =>
-        withResource(cv.any()) { isAny =>
-          if (isAny.isValid && isAny.getBoolean) {
+    def throwAnyNan(): Unit = {
+      withResource(values.isNan()) { valuesIsNan =>
+        withResource(valuesIsNan.any()) { anyNan =>
+          if (anyNan.isValid && anyNan.getBoolean) {
             throw RapidsErrorUtils.arithmeticOverflowError(errorMsg)
           }
         }
       }
     }
 
-    withResource(minValue) { minValue =>
-      throwIfAny(if (inclusiveMin) {
-        values.lessThan(minValue)
-      } else {
-        values.lessOrEqualTo(minValue)
-      })
+    def throwOutOfRange[T](minInput: T, minValue: T,
+                           maxInput: T, maxValue: T)
+                          (implicit ord: Ordering[T]): Unit = {
+      if (inclusiveMin && ord.compare(minInput, minValue) < 0 ||
+          !inclusiveMin && ord.compare(minInput, minValue) <= 0 ||
+          inclusiveMax && ord.compare(maxInput, maxValue) > 0 ||
+          !inclusiveMax && ord.compare(maxInput, maxValue) >= 0) {
+        throw RapidsErrorUtils.arithmeticOverflowError(errorMsg)
+      }
     }
 
-    withResource(maxValue) { maxValue =>
-      throwIfAny(if (inclusiveMax) {
-        values.greaterThan(maxValue)
-      } else {
-        values.greaterOrEqualTo(maxValue)
-      })
+    withResource(minValue) { minValue =>
```

Comment:
Let's rename `*Value` into something else like `*Bound` to reduce confusion. `minBound` and `maxBound` clearly say that they are the given limits. On the other hand, `minValue` and `maxValue` may be misunderstood as the min/max value of the input.

Reply:
I renamed *Value to *Bound. Thanks!